osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -0,0 +1,198 @@
+            {
+                background = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                },
+                scrollFlow = new BasicScrollContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    ScrollbarVisible = false,
+                    Child = new FillFlowContainer
+                    {
+                        AutoSizeAxes = Axes.Y,
+                        RelativeSizeAxes = Axes.X,
+                        Direction = FillDirection.Vertical,
+                        Children = new Drawable[]
+                        {
+                            new RankingsHeader
+                            {
+                                Anchor = Anchor.TopCentre,
+                                Origin = Anchor.TopCentre,
+                                Country = { BindTarget = country },
+                                Scope = { BindTarget = scope },
+                                Ruleset = { BindTarget = ruleset }
+                            },
+                            new Container
+                            {
+                                RelativeSizeAxes = Axes.X,
+                                AutoSizeAxes = Axes.Y,
+                                Children = new Drawable[]
+                                {
+                                    contentPlaceholder = new Container
+                                    {
+                                        Anchor = Anchor.TopCentre,
+                                        Origin = Anchor.TopCentre,
+                                        AutoSizeAxes = Axes.Y,
+                                        RelativeSizeAxes = Axes.X,
+                                        Margin = new MarginPadding { Vertical = 10 }
+                                    },
+                                    loading = new DimmedLoadingLayer(),
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colour)
+        {
+            Waves.FirstWaveColour = colour.Green;
+            Waves.SecondWaveColour = colour.GreenLight;
+            Waves.ThirdWaveColour = colour.GreenDark;
+            Waves.FourthWaveColour = colour.GreenDarker;
+
+            background.Colour = OsuColour.Gray(0.1f);
+        }
+
+        protected override void LoadComplete()
+        {
+            country.BindValueChanged(_ => redraw(), true);
+            scope.BindValueChanged(_ => redraw(), true);
+            ruleset.BindValueChanged(_ => redraw(), true);
+            base.LoadComplete();
+        }
+
+        public void ShowCountry(Country requested)
+        {
+            if (requested == null)
+                return;
+
+            Show();
+
+            if (country.Value?.FlagName == requested.FlagName)
+                return;
```

Comment:
This guard feels wrong and kind of defeats the point of `country` being a bindable. I'd probably rather have `Country` implement `IEquatable<Country>`, in which case this goes away magically and you get the same semantics.